### PR TITLE
fix: updated installation steps in the README file

### LINKED ISSUE #363: Omitted step in installation procedure in the README file
**Describe the bug**
The README file doesn't indicate how to rain check if npm have been installed in the local device and the steps to take. This could cause delay in the workflow as this minute detail could be overlooked.

**To Reproduce**

1. Go to the repository
2. Paste the following command on your local terminal git clone https://github.com/PalisadoesFoundation/talawa-docs.git
3. run npm install -g npm

**Expected behavior**
npm -g installed successfully

**Actual behavior**
npm: command not found

**Screenshots**
Screen Shot 2023-03-07 at 20 04 47

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by faveee**
**What kind of change does this PR introduce?**
Refactoring

**Issue Number:** #363 

Fixes #363 

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**
Screen Shot 2023-03-08 at 13 47 18

**If relevant, did you update the documentation?**
I updated the README file

**Summary**
Yesterday while trying to run the application on my local device, i faced this issue which lasted over 30mins. Having installed node and npm on my local device i skipped the node installation step only to get error while trying to run npm install -g npm. The error message stated that the command was not found. I proceeded to seek for help on slack channel but it was seeming impossible. Someone later suggested i confirmed if npm was installed with the following command npm -v. Apparently, i had somehow uninstall this at some certain point which i wasn't aware of. This PR can easily save people the stress.

**Does this PR introduce a breaking change?**
No, this change does not create a migration path for existing applications

**Other information**
All steps are gotten from the official sites respectively.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-docs/blob/develop/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by faveee** [comment]
@Cioppolo14 Kindly review the pull request and inform me if any improvement needed. Thank you!

File: README.md [hunk 1]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
 The steps are simple:
 
 1. Clone this repository
-2. [Install `yarn` on your system using these instructions](https://classic.yarnpkg.com/en/docs/install)
-3. Follow these steps next
+2. Install Yarn on your system following these instructions
+
+### Install Yarn on Windows Using NPM
```
Comment (palisadoes) [inline README.md:64]: Add a section or bullet point about the docusaurus prerequisites with a link to https://docusaurus.io/docs/installation where they are listed.

File: README.md [hunk 2]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
...
+
+- Install Yarn by running the given command in your terminal
+
+```terminal
+$ brew install yarn
+```
+
+### Install Yarn on Linux Using NPM
+
+- Run the following command in your terminal to install Node and NPM respectively. [Confirm your Linux distro and it's command prompt](https://linuxconfig.org/install-npm-on-linux)
```
Comment (palisadoes) [inline README.md]: Use a link from the yarn website. That will always be up to date.

File: README.md [hunk 3]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
...
+
+```terminal
+$ sudo apt install nodejs
+$ sudo apt install npm
+```
+- Verify installation
```
Comment (palisadoes) [inline README.md]: Be consistent use numbers versus bullets

File: README.md [hunk 4]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
...
+
+```terminal
+$ yarn -version
+```
+
+### Install Yarn on macOS Using NPM
+
+- [Click on the macOS Installer option to download the .pkg installer](https://nodejs.org/en/download/)
+- Run the Node.js installer
+- Verify Node.js Installation by running the following command in your terminal
```
Comment (palisadoes) [inline README.md]: Use numbers, not bullets. People can reference point `#4` which is easier to find.

File: README.md [hunk 5]
```diff
@@ -59,8 +142,71 @@ $ yarn add docusaurus
 The steps are simple:
 
 1. Clone this repository
-2. [Install `npm` on your system using these instructions](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
-3. Follow these steps next
+2. Install NPM on your system using these instructions
+
+### Install NPM on Windows
+
+- [Download the Windows installer](https://nodejs.org/en/download/)
```
Comment (palisadoes) [inline README.md]: Numbers vs bullets again.

File: README.md [hunk 6]
```diff
@@ -59,8 +142,71 @@ $ yarn add docusaurus
...
+```
+- Update your NPM version
+
+```terminal
+$ sudo npm install npm --global
+```
+
+### Install NPM on macOS Using Homebrew
+
+- Install Node and NPM by running the given command in your terminal
```
Comment (palisadoes) [inline README.md]: Numbers versus bullets

File: README.md [hunk 7]
```diff
@@ -59,8 +142,71 @@ $ yarn add docusaurus
...
+$ npm -v
+```
+- Update your NPM version
+
+```terminal
+$ sudo npm install npm --global
+```
+
+### Install NPM on macOS Using Homebrew
+
```
Comment (palisadoes) [inline README.md:202]: Please add introductory text below each heading. It makes the flow better.

File: README.md [hunk 8]
```diff
@@ -59,8 +142,71 @@ $ yarn add docusaurus
...
+- After choosing the path, double-click to install. Then give access to run the application
+- Check Node.js and NPM version by running the following command in your terminal
+
+```terminal
+$ node -v
+$ npm -v
+```
+
+### Install NPM on macOS Using .pkg Installer
+
```
Comment (palisadoes) [inline README.md:185]: Please add introductory text below each heading. It makes the flow better.

File: README.md [hunk 9]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
...
+### Install Yarn on macOS Using Homebrew
+
+- Install Yarn by running the given command in your terminal
+
+```terminal
+$ brew install yarn
+```
+
+### Install Yarn on Linux Using NPM
+
```
Comment (palisadoes) [inline README.md:127]: Please add introductory text below each heading. It makes the flow better.

File: README.md [hunk 10]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
...
+```
+
+- Verify Yarn Installation
+
+```terminal
+$ yarn --version
+```
+
+### Install Yarn on macOS Using Homebrew
+
```
Comment (palisadoes) [inline README.md:118]: Please add introductory text below each heading. It makes the flow better.

File: README.md [hunk 11]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
...
+- After choosing the path, double-click to install. Then give access to run the application
+
+- Check Yarn installation
+
+```terminal
+$ yarn -version
+```
+
+### Install Yarn on macOS Using NPM
+
```
Comment (palisadoes) [inline README.md:95]: Please add introductory text below each heading. It makes the flow better.

File: README.md [hunk 12]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
...
+```
+
+- Check Yarn installation
+
+```terminal
+$ yarn -version
+```
+
+### Install Yarn on Windows Using msi File
+
```
Comment (palisadoes) [inline README.md:83]: Please add introductory text below each heading. It makes the flow better.

File: README.md [hunk 13]
```diff
@@ -45,8 +45,91 @@ Installation will vary depending on whether you use the `yarn` or `npm` packages
 The steps are simple:
 
 1. Clone this repository
-2. [Install `yarn` on your system using these instructions](https://classic.yarnpkg.com/en/docs/install)
-3. Follow these steps next
+2. Install Yarn on your system following these instructions
+
+### Install Yarn on Windows Using NPM
+
```
Comment (palisadoes) [inline README.md:66]: Please add introductory text below each heading. It makes the flow better.

**Review by palisadoes — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (faveee) [reply README.md:64]: @palisadoes I used a Note option and bullet point to indicate the docusaurus requirements respectively. I have also implemented the other changes. Please check and let me know of any improvement i need to make.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**